Merge 1.7.3.x

eternoendless opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Merges 1.7.3.x into develop
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | -

This PR also makes some small changes to the way tests are launched. Check out this commit for more informaition: https://github.com/PrestaShop/PrestaShop/pull/8811/commits/41b44750af0bc761f161a84f54e2ac21a07fb49c

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/8811)
<!-- Reviewable:end -->

fouratachour committed:
Working on High test - category

fouratachour committed:
Code refactoring

fouratachour committed:
Travis configuration

fouratachour committed:
Working on README file

fouratachour committed:
Create attribute - high test

fouratachour committed:
Code refactoring

fouratachour committed:
Update attribute - high test

fouratachour committed:
Delete attribute/attribute value - high test

fouratachour committed:
Code Refactoring

fouratachour committed:
Create feature - high test

fouratachour committed:
Update feature - high test

fouratachour committed:
Delete feature - high test

fouratachour committed:
Deactivate product list - high test

fouratachour committed:
Activate product list - high test

fouratachour committed:
Duplicate product list - high test

fouratachour committed:
Code refactoring

fouratachour committed:
Code refactoring

fouratachour committed:
Add Brand and Brand Address

fouratachour committed:
Change the quantity for a product group

fouratachour committed:
Change the quantity for a single product

fouratachour committed:
Check product quantity movements

fouratachour committed:
Modify and check order quantity movement

fouratachour committed:
Code refactoring - rebase

fouratachour committed:
Adding onboarding function

fouratachour committed:
Create standard product

fouratachour committed:
Create pack of product/virtual product/product with combination

fouratachour committed:
Add check product BO to all type of product

fouratachour committed:
working on 1.7.3.0beta version

fouratachour committed:
Create cutomer and address

fouratachour committed:
Code refactoring

fouratachour committed:
Add default product sort and category radio button test

fouratachour committed:
Code refactoring

fouratachour committed:
Fixing category name feature

fouratachour committed:
Adding the onBoarding

fouratachour committed:
Adding product unique ID

fouratachour committed:
Adding create BO order and change invoice test

fouratachour committed:
Add check Invoice / delivery Invoice document

fouratachour committed:
Add create order in FO and check it in BO

fouratachour committed:
Migration of Create standard product

kpodemski committed:
IN: fix store hours migration

fouratachour committed:
Adding the new command and code refactoring

alexalouit committed:
BO: add missing roles

fouratachour committed:
Selector Refactoring

Commit:
Refactoring code of attribute,feature and create product

Quetzacoalt91 committed:
Update Translation Bundle version in composer

Quetzacoalt91 committed:
Merge pull request #8591 from Quetzacoalt91/transbundle-1.7.2.x

Update Translation Bundle version in composer

fouratachour committed:
Adding checkText Function to order and product

fouratachour committed:
Code refactoring

fouratachour committed:
Code Refactorng / BOOM 4361

fouratachour committed:
Code Refactoring

Commit:
Refactoring code of manufacturer

Commit:
Refactoring code of catalogbulkaction

Commit:
Check picture and quantity when generated combination

fouratachour committed:
Install Zip

fouratachour committed:
Add cross selling module test

fouratachour committed:
Modify and Check the edited translation

mbadrani committed:
rename the high test repo

fouratachour committed:
Local Installation

Commit:
Refactoring code of category

Commit:
Refactoring code of customer and address

eternoendless committed:
Include selenium standalone as a dependency & add scripts

Quetzacoalt91 committed:
Update docker-compose regarding new functional tests

fouratachour committed:
Installation in language = country / Installation in language <> country

eternoendless committed:
Add CssJanus fork and remove localize-fixture

eternoendless committed:
Move RTL CSS file generation into the core

eternoendless committed:
Fix missing slash in path

eternoendless committed:
Fix regeneration

fouratachour committed:
AutoUpgrade

eternoendless committed:
Enable Babel and UglifyJS for the new theme

fouratachour committed:
Rollback

fouratachour committed:
Adding module partnaire

fatmaBouchekoua committed:
TE: Tests refactoring

fatmaBouchekoua committed:
TE: Update scripts

fatmaBouchekoua committed:
TE: Update Readme file

fatmaBouchekoua committed:
TE: Fix order selectors

eternoendless committed:
Keep license comments in uglified javascript

eternoendless committed:
Build assets

eternoendless committed:
Remove console.log

fatmaBouchekoua committed:
TE: Refactor order's scripts

eternoendless committed:
Update UI Kit to 1.0.0-beta-4

toutantic committed:
Merge pull request #8657 from eternoendless/clean-rtl-dependencies

Integrate RTL CSS generation in the core

Commit:
Display all products, create account, create order with account

Quetzacoalt91 committed:
Merge pull request #8659 from eternoendless/webpack-env

Re-enable uglify on the new theme

toutantic committed:
Merge pull request #8670 from Quetzacoalt91/cherry-pick-8556

Fix store hours migration

fouratachour committed:
Upgrade review code + Readme file

Commit:
Check all products page/Check products number in FO

Commit:
Enable and disable shop

Commit:
Open all menu links

Commit:
OnBoarding + refactoring all menu

Danoosh committed:
Some RTL fixtures

Danoosh committed:
Add module RTL stylesheets generator

Commit:
Adding order common functions

eternoendless committed:
Extract RTL stylesheet processor

fatmaBouchekoua committed:
TE: Add regular campaign

Commit:
Search module

eternoendless committed:
Fix help sidebar on firefox
+ Fixed offscreen employee menu

eternoendless committed:
Removed unneeded ternary

eternoendless committed:
Fixed spinner button

eternoendless committed:
Build assets

eternoendless committed:
Merge pull request #8697 from eternoendless/add-missing-roles

Added Back Office missing roles

eternoendless committed:
Fixed wrong module path

eternoendless committed:
Merge pull request #8691 from iPresta/1.7.3.module-rtl

Add RTL generation for modules and some rtl fixtures

eternoendless committed:
Moved CSS out of many vue components to allow RTLization

+ Fixed many graphical glitches

eternoendless committed:
Fix text button background on Firefox

eternoendless committed:
Build assets

Commit:
Sort product

Quetzacoalt91 committed:
Merge pull request #8671 from eternoendless/update-ui-kit

Fix visual regressions on Firefox

eternoendless committed:
Add support for arabic numbers

eternoendless committed:
Merge pull request #8702 from eternoendless/BOOM-4319-fix-rtl

Fix RTL visual glitches

Quetzacoalt91 committed:
Merge pull request #8704 from eternoendless/parse-arabic-numbers

Support Arabic numbers in the Product page

eternoendless committed:
Update native modules version

fouratachour committed:
Sort module by name, Increasing price, Decreasing price, popularity

fouratachour committed:
Create employee

fouratachour committed:
delete employee

toutantic committed:
BO: update translation strings catalog for 1.7.3

Commit:
Open all controllers

toutantic committed:
Update composer.lock for prestashop/* modules dependencies

toutantic committed:
BO: update catalog for 1.7.3

toutantic committed:
BO: update catalog for 1.7.3

toutantic committed:
BO: update catalog for 1.7.3

toutantic committed:
BO: update catalog for 1.7.3

fatmaBouchekoua committed:
TE: Add headless option

toutantic committed:
Merge pull request #8713 from PrestaShop/translation1.7.3

String to translate for 1.7.3

fatmaBouchekoua committed:
TE: Add directory for screenshots

alegout committed:
CO: Fix missing CLDR supplemental files

toutantic committed:
Merge pull request #8719 from alegout/release-tool

Fix missing CLDR supplemental directory

Commit:
Check the discount on Product page

toutantic committed:
Delete uneeded translation file (Case issue)

LittleBigDev committed:
Fixed combination images selection

LittleBigDev committed:
Build assets

toutantic committed:
New demo products

toutantic committed:
Update composer.lock for demo products update on modules

Commit:
Check the Impact on price (tax incl.)

toutantic committed:
Update logo for demo products update

toutantic committed:
remove uneeded files & fix module change on orders.xml

toutantic committed:
Fix translations for language other than en & fr

toutantic committed:
Merge pull request #8737 from LittleBigDev/BOOM-4745-combination-images

Fixed combination images selection

toutantic committed:
demo product add missing relations between product declinaison and images

Commit:
Check that the shopping cart dosen't allow checkout of zero quantity and inactive products

fatmaBouchekoua committed:
TE: Test the impact of decimals with comma

fouratachour committed:
BOOM-4448

toutantic committed:
Fix unit test with new demo products

fatmaBouchekoua committed:
TE: Check changed quantity on cart page

toutantic committed:
Fix " VS ' in yaml configuration file

toutantic committed:
MockContext to fix unit test that was failling when run after another

toutantic committed:
new demo products:  fix some relations

toutantic committed:
Fix unit test-admin with new demo products

fouratachour committed:
Problem showing discount in cart

toutantic committed:
Fix unit test with new demo products

toutantic committed:
New demo products update id supplier

toutantic committed:
Fix unit test with new demo products

eternoendless committed:
Fix selenium fixtures

toutantic committed:
Update composer.lock with latest versions of PrestaShop modules

toutantic committed:
Update translations catalog

fouratachour committed:
Install shop with english language and iran country

toutantic committed:
New demo products fix orders

toutantic committed:
Fix selenium test set the customizable product id to 1 like in the prepare-shop.php script

fouratachour committed:
Check ordering a quantity of product out of stock

Commit:
Refactoring code

eternoendless committed:
Merge pull request #8739 from PrestaShop/demoProduct

Demo products

fouratachour committed:
Check the prestashop module installed after the upgrade

alegout committed:
BO: Fix directory listing

alegout committed:
BO: Fix directory listing

fouratachour committed:
Use Gulp to concatenate the css and js files in the html file

tomlev committed:
use json_encode instead of serialize

fatmaBouchekoua committed:
TE: Add script sanity-check

fatmaBouchekoua committed:
TE: Update Readme file

fatmaBouchekoua committed:
TE: Update tests with new demo data

tomlev committed:
fix cart delivery option

fouratachour committed:
Adding Discount folder

fouratachour committed:
Rename product test

fatmaBouchekoua committed:
TE: Add new campaign for install and upgrade tests

fatmaBouchekoua committed:
TE: Fix tests

tomlev committed:
replace serialize with json_encode

eternoendless committed:
Merge pull request #8772 from tomlev/BOOM-4804-1

replace serialize with json_encode

eternoendless committed:
Merge pull request #8761 from tomlev/BOOM-4804

use json_encode instead of serialize

eternoendless committed:
Merge pull request #8755 from alegout/boom-4803

Fix directory listing

eternoendless committed:
// Changelog 1.7.2.5

eternoendless committed:
Merge branch '1.7.2.x' into 1.7.3.x

eternoendless committed:
Fix typo in 1.7.2.5 release date

eternoendless committed:
Add customization field to customizable demo product

eternoendless committed:
Fix undefined property in CustomizationField

eternoendless committed:
Enable customization by default

eternoendless committed:
Update theme preview

Quetzacoalt91 committed:
Merge pull request #8794 from eternoendless/customization-fixture

Fix customizable demo product without customizable field

Quetzacoalt91 committed:
Merge pull request #8796 from eternoendless/update-theme-preview

Update theme preview image

eternoendless committed:
Merge pull request #8802 from fatmaBouchekoua/FT-1.7.3.x

Add functional tests on 1.7.3.x

eternoendless committed:
Merge pull request #8777 from eternoendless/fix-directory-BOOM-4803

Fix directory listing

eternoendless committed:
Update composer dependencies

eternoendless committed:
// Changelog 1.7.3.0

eternoendless committed:
Merge '1.7.3.x' into develop

Quetzacoalt91 commented:
Issue in Travis:

```
$ composer phpunit-legacy
> Tests\PrestaShopBundle\Utils\Database::createTestDB
> @php -d date.timezone=UTC ./vendor/bin/phpunit -c tests/phpunit.xml
PHPUnit 5.7.27 by Sebastian Bergmann and contributors.
..............................................................E  63 / 537 ( 11%)
EE.........S................................................... 126 / 537 ( 23%)
............................................................... 189 / 537 ( 35%)
............................................................... 252 / 537 ( 46%)
............................................................... 315 / 537 ( 58%)
............................................................... 378 / 537 ( 70%)
............................................................... 441 / 537 ( 82%)
............................................................... 504 / 537 ( 93%)
.................................                               537 / 537 (100%)
Time: 1.87 minutes, Memory: 122.00MB
There were 3 errors:
1) Tests\Unit\Classes\CartTest::test_getOrderTotal_Round_Line_When_No_Tax
Error: Class 'PrestaShop\PrestaShop\Tests\Unit\ContextMocker' not found
/home/travis/build/PrestaShop/PrestaShop/tests/Unit/Classes/CartTest.php:93
2) Tests\Unit\Classes\CartTest::test_getOrderTotal_Round_Total_When_No_Tax
Error: Class 'PrestaShop\PrestaShop\Tests\Unit\ContextMocker' not found
/home/travis/build/PrestaShop/PrestaShop/tests/Unit/Classes/CartTest.php:93
3) Tests\Unit\Classes\CartTest::test_getOrderTotal_Round_Item_When_No_Tax
Error: Class 'PrestaShop\PrestaShop\Tests\Unit\ContextMocker' not found
/home/travis/build/PrestaShop/PrestaShop/tests/Unit/Classes/CartTest.php:93
ERRORS!
Tests: 537, Assertions: 882, Errors: 3, Skipped: 1.
Script @php -d date.timezone=UTC ./vendor/bin/phpunit -c tests/phpunit.xml handling the phpunit-legacy event returned with error code 2
The command "composer phpunit-legacy" exited with 2
```

eternoendless committed:
Fix SQL error when Shop context is null

eternoendless committed:
Fix not being able to launch sf-tests from IDE

eternoendless committed:
Fix wrong path to symfony app

eternoendless committed:
Fix CartTest failing because of unexpected global context

eternoendless committed:
Pass KERNEL_DIR as an environment variable instead of hardcoding it in phpunit configuration file

There's a reason for doing this:
If you run tests from composer, you KERNEL_DIR will be "app". But if you run tests from your IDE by lancing a phpunit xml configuration file, it should be "../app" since xml files are in the "tests" directory.

This setting cannot be inside the xml file because phpunit will prefer its config file to whatever environment variable you set. If you need to run your tests from the IDE, make sure you set up the correct KERNEL_DIR environment variable on your run configuration.
